packages/y-indexeddb/src/provider.ts

```diff
@@ -29,12 +30,24 @@ const createDatasource = ({
   dbName: string;
   mergeCount?: number;
 }) => {
+  let status: Status = {
+    type: 'idle',
+  };
+  const callbackSet = new Set<() => void>();
+  const changeStatus = (newStatus: Status) => {
+    status = newStatus;
+    callbackSet.forEach(cb => cb());
+  };
+
   const dbPromise = openDB<BlockSuiteBinaryDB>(dbName, dbVersion, {
     upgrade: upgradeDB,
   });
   const adapter = {
     queryDocState: async (guid, options) => {
       try {
+        changeStatus({
```

Comment:
I am not sure but it seems the way setting status may has race condition issues

Reply:
I think you are right. we need a mutex lock here

Reply:
I simulate a stack, it works great